### PR TITLE
fix: schema generate for tuple only with default

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
Generate the wrong schema for tuple only with default

**Related Issue:**
#1567

### CODE REVIEW

File: pkg/templates/loader/terraform_loader_test.go [hunk 1]
```diff
@@ -18,13 +20,187 @@ var mockInfo = &openapi3.Info{
...
+											WithUIGroupOrder("Basic").
+											Export(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedError: false,
```
Comment (thxCode) [inline pkg/templates/loader/terraform_loader_test.go:202]: can we describe the data in YAML(to ensure the structure), and then marshal it as openapi schema before each test run?

File: pkg/templates/openapi/operation.go [hunk 2]
```diff
@@ -94,3 +98,36 @@ func UnionSchema(s1, s2 *openapi3.Schema) (*openapi3.Schema, error) {
...
+	s2Byte, err := s2.MarshalJSON()
+	if err != nil {
+		return false, err
+	}
+
+	return bytes.Equal(s1Byte, s2Byte), nil
+}
+
+// MushSchemaEqual checks if s1 and s2 are equal, just log the error.
+func MushSchemaEqual(s1, s2 *openapi3.Schema) bool {
```
Comment (gitlawr) [inline pkg/templates/openapi/operation.go:126]: Is it must?

File: pkg/templates/translator/terraform_translator.go [hunk 3]
```diff
@@ -141,26 +149,24 @@ func (t TerraformTranslator) SchemaOfOriginalType(tp any, opts Options) *openapi
...
-			if diffSchema {
-				s.WithItems(openapi3.NewObjectSchema()).
-					WithLength(int64(len(ts)))
+			if !schemaEqual {
+				// NB: if the tuple items type are different, we use object schema to represent.
+				s.WithItems(openapi3.NewObjectSchema())
 			} else {
-				s.WithItems(refs[0]).
-					WithLength(int64(len(ts)))
+				s.WithItems(refs[0])
```
Comment (gitlawr) [inline pkg/templates/translator/terraform_translator.go:169]: Why remove WithLength?

Comment (aiwantaozi) [reply pkg/templates/translator/terraform_translator.go:169]: WithLength will add limit for maxLength and minLength

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by thxCode — APPROVED** [review]
LGTM

**Pull request merged**